xskillscore/core/comparative.py

```diff
@@ -280,39 +295,79 @@ def mae_test(
     """
     if isinstance(dim, str):
         dim = [dim]
+    elif dim is None:
+        dim = []
+
     if time_dim in dim:
         raise ValueError("`dim` cannot contain `time_dim`")
 
     msg = f"`alpha` must be between 0 and 1 or `return_p`, found {alpha}"
-    if isinstance(alpha, str):
-        if alpha != "return_p":
+    if isinstance(alpha, (str, int)) and alpha != "return_p":
+        raise ValueError(msg)
+
+    if isinstance(alpha, float) and not (0 < alpha < 1):
+        raise ValueError(msg)
+
+    if observations is not None and isinstance(metric, str):
+        # Compare the forecasts and observations using metric
+        valid_metrics = [
+            "me",
+            "rmse",
+            "mse",
+            "mae",
+            "median_absolute_error",
+            "smape",
+        ]
+        if metric not in valid_metrics:
+            msg = (
+                f"`metric` should be a valid distance metric function, found {metric}."
+                + " Valid metrics are:\n"
+                + ", ".join(valid_metrics)
+            )
             raise ValueError(msg)
-    elif isinstance(alpha, float):
-        if not 0 < alpha < 1:
+
+        err_func = getattr(dm, metric)
+        if dim is not None and "dim" in kwargs:
+            kwargs.pop("dim")
+
+        params = inspect.signature(err_func).parameters
+        missing_args = [
+            p
+            for p, v in params.items()
+            if v.default == inspect._empty and p not in kwargs and p not in ["a", "b"]
+        ]
+        if len(missing_args) > 2:
+            msg = (
+                f"The following positional arguments for {metric} are missing:\n"
+                + ", ".join(missing_args)
+            )
```

Comment:
nice. we could try to use this for the metrics themselves @raybellwaves

Reply:
yeah i'm not familiar with inspect. looks good.